swig/cpp/tests/changes.cpp

```diff
@@ -0,0 +1,219 @@
+#include <iostream>
+#include <memory>
+#include <cassert>
+#include <cstring>
+#include <unistd.h>
+
+#include "Session.h"
+
+#define MAX_LEN 100
+
+using namespace std;
+
+const char *module_name = "swig-test";
+const int LOW_BOUND = 10;
+const int HIGH_BOUND = 20;
+const char *xpath_if_fmt = "/swig-test:cpp-operations/test-get[name='%s']/%s";
+
+char *
+get_test_name(int i)
+{
+    char buf[MAX_LEN] = {0};
+    sprintf(buf, "test-cpp-%d", i);
+    return strdup(buf);
+}
+
+const char *
+get_xpath(char *test_name, char *node_name)
```

Comment:
I'm a C++ programmer, so please take this into perspective when reading this comment. This is a unit test for a C++ part of a program. Even if C code was more efficient, it doesn't matter because this is a test. In C++, this would become:

```c++
std::string get_xpath(const std::string &test_name, const std::string &node_name)
{
  return "/swig-test:cpp-operations/test-get[name='" + test_name + "']/" + node_name;
}
```

...and see a couple of comments below for usage.